src/Markup/Avalonia.Markup.Xaml.Loader/CompilerExtensions/Transformers/AvaloniaXamlIlBindingPathTransformer.cs

```diff
@@ -123,7 +123,12 @@ IEnumerable<XamlPropertyAssignmentNode> getResourceValues(IXamlAstNode node)
                 };
 
                 var selfType = context.ParentNodes().OfType<XamlAstConstructableObjectNode>().First().Type.GetClrType();
-                
+
+                if (context.GetAvaloniaTypes().MultiBinding.IsAssignableFrom(selfType))
```

Comment:
It should be a loop to handle nested `MultiBinding`s.
(Granted, it's probably a rare case, but they work in reflection mode.)

Reply:
I don't see as much problem with nested MultiBinding, but more with other markup extensions in general.
Like, Binding inside of OnPlatform extension. Which wouldn't be as rare use case.

Reply:
More generic fix can be added later though. Just to keep in mind.

Reply:
`OnPlatform`/`OnFormFactor` both already work with this PR, because they're transformed into special nodes which aren't `XamlAstConstructableObjectNode`. I couldn't find any other case that the nested `MultiBinding` one, which is easy to handle. If there are others I'm missing, we can indeed add a more general solution later.

Reply:
Yeah, other would be MarkupExtension implementations that don't have special transformer support from the compiler.

Reply:
I think nested MultiBindings could be handled by doing this?
```csharp
  var selfType = context.ParentNodes().OfType<XamlAstConstructableObjectNode>()
      .Select(n => n.Type.GetClrType())
      .Where(t => !context.GetAvaloniaTypes().MultiBinding.IsAssignableFrom(t))
      .First();
```